arbnode/dataposter/state/state.go

```diff
@@ -0,0 +1,70 @@
+package state
+
+import (
+	"context"
+	"math/big"
+	"sync"
+
+	"github.com/offchainlabs/nitro/arbnode/dataposter/storage"
+)
+
+// lint:require-exhaustive-initialization
+type InternalState struct {
+	mutex       sync.Mutex
+	lockedState LockedInternalState
+}
+
+func NewInternalState(queue QueueStorage) *InternalState {
+	return &InternalState{
+		mutex: sync.Mutex{},
+		lockedState: LockedInternalState{
+			LastBlock:  big.NewInt(0),
+			Balance:    big.NewInt(0),
+			Nonce:      0,
+			Queue:      queue,
+			ErrorCount: make(map[uint64]int),
+		},
+	}
+}
+
+func (s *InternalState) Lock() *LockedInternalState {
```

Comment:
nit: for a few cases it could be nice to have something like this:  
```
func (s *InternalState) WithLock(fn func(st *LockedInternalState)) {
    s.mutex.Lock()
    defer s.mutex.Unlock()
    fn(&s.lockedState)
}
```